indico/modules/events/registration/models/registrations.py

```diff
@@ -595,7 +595,7 @@ def get_personal_data(self):
         personal_data = {}
         for data in self.data:
             field = data.field_data.field
-            if field.personal_data_type is not None and data.data:
+            if field.personal_data_type is not None and (data.data or data.filename):
                 personal_data[field.personal_data_type.name] = data.friendly_data
```

Comment:
Maybe in case of a picture field we should add a reference to the data object itself (since it contains `.open()` etc? That way you could use `get_personal_data()` e.g. when rendering the placeholder instead of having to manually loop over the data.

Reply:
If we use the new property, in render of PicturePlaceholder =>
picture_data = registration.picture_data

Reply:
Maybe not adding it to `get_personal_data` but just having `get_personal_picture` might be better then? (the filename in the personal data is pretty much useless)

Reply:
"the filename in the personal data is pretty much useless": I know :-)
it's just to mean there is a picture in the registration.

I don't get what you meant:
 add a method 'get_personal_picture' ? but what to return in picture for get_personal_data ?

Reply:
you're right, since `open()` is meant to be used in a `with` block that mathod would not be particularly useful. Maybe `get_personal_data_picture` that returns the actual RegistrationData for that field, sine this is where you have e.g. the `open` and `send` methods

Reply:
> get_personal_data_picture that returns the actual RegistrationData for that field
=> Ok,
but what would you like to get for picture in "get_personal_data" ?
(currently I just return the useless filename in personal data)

Reply:
I would be completely OK with skipping it altogether, or alternatively including the RegistrationData object there directly.

Actually, after looking at the code: Building the full personal data dict exactly as (in)expensive as just getting the RegistrationData for the picture, since you're probably going to iterate over all the personal data fields in either case. So maybe the "standalone" method is not necessary, and if you need the picture regdata you just use `.get_personal_data()['picture']`?